dev/com.ibm.ws.logging/src/com/ibm/ws/logging/internal/impl/ThrottleState.java

```diff
@@ -0,0 +1,98 @@
+/*******************************************************************************
+ * Copyright (c) 2025 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package com.ibm.ws.logging.internal.impl;
+
+import java.util.function.Supplier;
+
+class ThrottleState {
```

Comment:
Add a comment on what this ThrottleState class is doing, and explain the moving window concept.